packages/headless/src/modules/customer/services/index.ts

```diff
@@ -44,8 +44,8 @@ export class CustomerService {
     }
 
     async addCustomerNewAddress(customerId: string, address: CustomerAddress): Promise<AddCustomerNewAddressResponse> {
-        let customer = await this.customerRepo.findCustomer({ id: customerId });
-        if (!customer) return this.helper.serviceResponse.errorResponse(GetCustomerInformationErrorMessages.CUSTOMER_NOT_FOUND, null, HttpStatus.BAD_REQUEST);
+        const doesDefaultAddressExists = address.isDefault && await this.customerRepo.findCustomer({ id: customerId, 'addresses.isDefault': true });
+        if (doesDefaultAddressExists) return this.helper.serviceResponse.errorResponse(AddCustomerNewAddressErrorMessages.ALREADY_DEFAULT_ADDRESS_EXISTS, null, HttpStatus.BAD_REQUEST);
```

Comment:
don't block like this. in this case customer have to remove the previous default. lets discuss next day

Reply:
This service function should add a new address. So, if this customer already has a default address, we will prevent them from adding a new one. If the customer has a previous default address, then he can either delete this address or update the isDefault property to false.